src/Tempest/Database/src/ConnectionInitializer.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database;
+
+use Tempest\Container\Container;
+use Tempest\Container\Initializer;
+use Tempest\Container\Singleton;
+
+final class ConnectionInitializer implements Initializer
+{
+    private static Connection|null $instance = null;
+
+    #[Singleton]
+    public function initialize(Container $container): Connection
+    {
+        // Reuse same connection instance in unit tests
```

Comment:
If this is only meant to be used in tests, can we ensure we are actually running in tests? We can check the current environment from the Kernel I believe

Reply:
This was a cary over from the PDOInitializer which had the same. But yeah, good suggestion. I will make it such that it only "caches" when in testing. This is more of a test speed+memory improvement than anything else.

Reply:
Implemented the check.